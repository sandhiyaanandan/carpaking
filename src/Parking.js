import React from 'react';
import {TOTAL_AVAILABLE_SPACE} from './constants';
import {useHistory } from "react-router-dom";
function Parking()
{
	let parkingBox = [];
	let no_available_space = TOTAL_AVAILABLE_SPACE;
	const history = useHistory(); //used to navigate to other page or component
	var alertMsg="";
	const isParkingSelected = (e) => {	
		var cUsrDetails =null;
		let isCarParked = document.activeElement.style;
		var curUser = localStorage.getItem("currentUser");
		if(curUser !== null) {
			cUsrDetails = JSON.parse(curUser); 
		}
		if(isCarParked.backgroundColor === "" || isCarParked.backgroundColor === "white") {
			if( no_available_space > 0 ) {
				if (alertMsg === "Car In") {
					alert("Your car is already parked..Thank you");
					return;
				} else {
					alertMsg = "Car In";
					isCarParked.backgroundColor = "yellow";
				}
			}
				cUsrDetails.parkingPos =  e.target.value;
				localStorage.setItem('currentUser', JSON.stringify(cUsrDetails));
			 no_available_space--;
		} else {
			//Get carnumber from user and check in local storage...if available change color to white
			var userCarNumber  = prompt("Please enter your carNumber");
			if ( cUsrDetails.carNumber === userCarNumber )
			{
				cUsrDetails.parkingPos = -1;
				alertMsg = "Car OUT";
				isCarParked.backgroundColor = "white";
				if( no_available_space >= 0 ) {
					no_available_space++;
				}
			} else {
				alertMsg = "Try again. Please provide registered carNumber";
				history.push("/Login");
			}
		}
		alert(alertMsg);
		var regiteredDetails = localStorage.getItem("regDetails");
		let custDetails =[];
		if(regiteredDetails !== null) {
			custDetails = JSON.parse(regiteredDetails); 
			if (custDetails.length > 0 )
			{
				custDetails = custDetails.map((cDetail) => {
					if(cDetail.carNumber === cUsrDetails.carNumber) {
						return {
							...cDetail,
							parkingPos: cUsrDetails.parkingPos
						}
					}
					return cDetail;
				})
			} else {
				custDetails.push(cUsrDetails);
			}
		}
		localStorage.setItem('regDetails', JSON.stringify(custDetails));
		localStorage.setItem('no_available_space', no_available_space);
	}
	
	const allUser = localStorage.getItem("regDetails");
	var posArr=0;
	if(allUser !== null) {
		const cDetails = JSON.parse(allUser); 
		if (cDetails.length > 0 )
		{
			posArr = (cDetails.map((detail)=> {
				let pos = parseInt(detail.parkingPos);
				return pos;
			}));
		}
	}

	for( let pos = 0; pos < TOTAL_AVAILABLE_SPACE; pos++)
	{
		if( pos % 5 === 0 )
			parkingBox.push(<br/>);
		if(posArr && ( posArr.indexOf(pos)!== -1))
		{
			parkingBox.push(<input type="button" style={{backgroundColor:'yellow'}} value={pos} className="box" onClick={isParkingSelected}/>);
		} else {
			parkingBox.push(<input type="button" value={pos} className="box" onClick={isParkingSelected}/>);
		}		
	}

	return(<div style={{backgroundColor:'red'}} data-test="Parking-Page-Header">
		<h1 className="pageHeader" style={{color:'white'}}>Parking Area</h1>
		<div className="parkingArea" >
			{parkingBox}
		</div>
	</div>)	
}
export default Parking;