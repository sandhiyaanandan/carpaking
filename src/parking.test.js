import React from 'react';
import { shallow } from 'enzyme';
import Parking from './parking';
import {findByTestAttr} from './../jestutils';
import toJson from 'enzyme-to-json';

const setUp =()=>{
    const component = shallow(<Parking />);
	return component;
};

jest.mock('react-router', () => ({     ...jest.requireActual('react-router'),    
 useHistory: () => ({ push: jest.fn() }) }));  
describe('Parking', ()=>{
	let wrapper;
	beforeEach(() =>{
		wrapper = setUp();
	});
	
	it('should render the Parking Component correctly', ()=>{   
    	expect(toJson(wrapper)).toMatchSnapshot();
  	});

	it('should render Parking component correctly', ()=>{
		let comp = findByTestAttr(wrapper, 'Parking-Page-Header');
		expect(comp.length).toBe(1);
	});
});




