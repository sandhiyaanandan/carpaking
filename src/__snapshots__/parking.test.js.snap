// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parking should render the Parking Component correctly 1`] = `
<div
  data-test="Parking-Page-Header"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <h1
    className="pageHeader"
    style={
      Object {
        "color": "white",
      }
    }
  >
    Parking Area
  </h1>
  <div
    className="parkingArea"
  >
    <br />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={0}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={1}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={2}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={3}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={4}
    />
    <br />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={5}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={6}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={7}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={8}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={9}
    />
    <br />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={10}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={11}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={12}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={13}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={14}
    />
    <br />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={15}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={16}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={17}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={18}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={19}
    />
    <br />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={20}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={21}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={22}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={23}
    />
    <input
      className="box"
      onClick={[Function]}
      type="button"
      value={24}
    />
  </div>
</div>
`;
